src/components/ui/Input/SquareTextInput.tsx

```diff
@@ -0,0 +1,101 @@
+'use client';
+import { CircleCrossIcon } from '@/assets/images/icons';
+import { useOutsideClick } from '@/hooks';
+import { cn } from '@/lib/tailwind/utils';
+
+import { VariantProps } from 'class-variance-authority';
+import { forwardRef, InputHTMLAttributes, useRef } from 'react';
+
+import { Button } from '../Button';
+import { Icon, iconVariants } from '../Icon';
+import { Input, inputVariants } from './Input';
+
+interface ISquareTextInputProps extends Omit<InputHTMLAttributes<HTMLInputElement>, 'type'> {
+  /**
+   * Function to clear the input value with cross button
+   */
+  handleClearInput?: () => void;
```

Comment:
handleClearInput is optional in here, and if it's not provided, cross button will not be displayed.